File: 0012-integer-to-roman/0012-integer-to-roman.cpp
#include <string>
using namespace std;

class Solution {
public:
    string intToRoman(int num) {
        string result = "";
        int values[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        string symbols[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        int i = 0;
        while (num > 0) {
            if (num >= values[i]) {
                num -= values[i];
                result += symbols[i];
            }
            else {
                i++;
            }
        }
        return result;
    }
};

File: 0012-integer-to-roman/0012-integer-to-roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        result = ""
        values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        symbols = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
        i = 0
        while num > 0:
            if num >= values[i]:
                num -= values[i]
                result += symbols[i]
            else:
                i += 1
        return result
